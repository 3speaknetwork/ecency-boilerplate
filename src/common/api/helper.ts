import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
