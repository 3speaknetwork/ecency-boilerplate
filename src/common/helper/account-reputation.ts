export default (input: string | number): number => {

  if(typeof input === 'string'){
    input = Number(input);
  }
  
  if (isFloat(input)) {
    return Math.floor(input);
  }
  
  if (input === 0) {
    return 25;
  }

  if (!input) {
    return input;
  }

  let neg = false;

  if (input < 0) neg = true;

  let reputationLevel = Math.log10(Math.abs(input));
  reputationLevel = Math.max(reputationLevel - 9, 0);

  if (reputationLevel < 0) reputationLevel = 0;

  if (neg) reputationLevel *= -1;

  reputationLevel = reputationLevel * 9 + 25;

  return Math.floor(reputationLevel);
};

function isFloat(n: string | number) {
  return Number(n) === n && n % 1 !== 0;
}
