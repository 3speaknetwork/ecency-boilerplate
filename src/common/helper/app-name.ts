interface ObjInput {
    name?: string
}

export default (input: string | null | undefined | ObjInput) => {
    if (!input) {
        return '';
    }

    if (typeof input === 'string') {
        return input;
    }

    if (typeof input === 'object' && input.name !== undefined) {
        return input.name;
    }

    return '';
};
