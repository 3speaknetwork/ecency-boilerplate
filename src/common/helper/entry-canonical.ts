import {Entry} from "../store/entries/types";

import apps from "@hiveio/hivescript/apps.json";

import defaults from "../../common/constants/defaults.json";

export default (entry: Entry): string | null => {
    if (entry.json_metadata?.canonical_url) {
        return entry.json_metadata?.canonical_url;
    }

    let scheme = `${defaults.base}/{category}/@{username}/{permlink}`;

    if (entry.json_metadata?.app) {
        const identifier = entry.json_metadata.app.split("/")[0];

        if (apps[identifier]) {
            scheme = apps[identifier].url_scheme;
        }
    }

    if (!scheme) {
        return null;
    }

    return scheme
        .replace("{category}", entry.category)
        .replace("{username}", entry.author)
        .replace("{permlink}", entry.permlink);
};
