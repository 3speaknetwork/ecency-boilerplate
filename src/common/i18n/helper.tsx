import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
