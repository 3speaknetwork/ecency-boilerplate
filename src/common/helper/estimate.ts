import { votingPower } from "../api/hive";
import { ActiveUser } from "../store/active-user/types";
import { DynamicProps } from "../store/dynamic-props/types";
import { Entry } from "../store/entries/types";
import parseAsset from "./parse-asset";

export const estimate = (
  entry: Entry,
  activeUser: ActiveUser,
  dynamicProps: DynamicProps,
  percent: number
): number => {
  if (!activeUser) {
    return 0;
  }

  const { fundRecentClaims, fundRewardBalance, base, quote } = dynamicProps;
  const { data: account } = activeUser;

  if (!account.__loaded) {
    return 0;
  }

  const sign = percent < 0 ? -1 : 1;
  const postRshares = entry.net_rshares;

  const totalVests =
    parseAsset(account.vesting_shares).amount +
    parseAsset(account.received_vesting_shares).amount -
    parseAsset(account.delegated_vesting_shares).amount;

  const userVestingShares = totalVests * 1e6;

  const userVotingPower = votingPower(account) * Math.abs(percent);
  const voteEffectiveShares = userVestingShares * (userVotingPower / 10000) * 0.02;

  // reward curve algorithm (no idea whats going on here)
  const CURVE_CONSTANT = 2000000000000;
  const CURVE_CONSTANT_X4 = 4 * CURVE_CONSTANT;
  const SQUARED_CURVE_CONSTANT = CURVE_CONSTANT * CURVE_CONSTANT;

  const postRsharesNormalized = postRshares + CURVE_CONSTANT;
  const postRsharesAfterVoteNormalized = postRshares + voteEffectiveShares + CURVE_CONSTANT;
  const postRsharesCurve =
    (postRsharesNormalized * postRsharesNormalized - SQUARED_CURVE_CONSTANT) /
    (postRshares + CURVE_CONSTANT_X4);
  const postRsharesCurveAfterVote =
    (postRsharesAfterVoteNormalized * postRsharesAfterVoteNormalized - SQUARED_CURVE_CONSTANT) /
    (postRshares + voteEffectiveShares + CURVE_CONSTANT_X4);

  const voteClaim = postRsharesCurveAfterVote - postRsharesCurve;

  const proportion = voteClaim / fundRecentClaims;
  const fullVote = proportion * fundRewardBalance;

  const voteValue = fullVote * (base / quote);

  if (sign > 0) {
    return Math.max(voteValue * sign, 0);
  }

  return voteValue * sign;
};
