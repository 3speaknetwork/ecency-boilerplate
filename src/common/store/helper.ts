import {Store} from "redux";

const getSymbolFromCurrency = require("currency-symbol-map");

import {AppState} from "./index";
import {ActiveUser, UserPoints} from "./active-user/types";
import {loginAct as loginActiveUser, updateAct as updateActiveUserAct} from "./active-user";

import {getAccount, getDynamicProps} from "../api/hive";
import {getPoints, usrActivity, getPromotedEntries} from "../api/private";
import {reloadAct as reloadUsers} from "./users";
import {reloadAct as reloadReblogs} from "./reblogs";
import {fetchedAct as loadDynamicProps} from "./dynamic-props";
import {fetchedAct as entriesFetchedAct} from "../../common/store/entries";
import {setCurrencyAct as setCurrency, muteNotificationsAct as muteNotifications} from "./global";

import {getCurrencyRate} from "../api/misc";

import currencies from "../constants/currencies.json";

import * as ls from "../../common/util/local-storage";

export const activeUserMaker = (name: string, points: string = "0.000", uPoints: string = "0.000"): ActiveUser => {
    return {
        username: name,
        data: {name},
        points: {
            points,
            uPoints
        }
    }
}

export const activeUserUpdater = async (store: Store<AppState>) => {
    const state = store.getState();
    if (state.activeUser) {
        const {username} = state.activeUser;

        let account;
        try {
            account = await getAccount(username);
        } catch (e) {
            return;
        }

        let points: UserPoints;
        try {
            const p = await getPoints(username);
            points = {points: p.points, uPoints: p.unclaimed_points};
        } catch (e) {
            points = {
                points: "0.000",
                uPoints: "0.000"
            };
        }

        store.dispatch(updateActiveUserAct(account, points));
    }
};

export const clientStoreTasks = (store: Store<AppState>) => {
    // Initial state from browser's local storage
    store.dispatch(reloadUsers());
    store.dispatch(loginActiveUser());
    store.dispatch(reloadReblogs());

    // Load dynamic props
    getDynamicProps().then((resp) => {
        store.dispatch(loadDynamicProps(resp));
    });

    // Update active user in interval
    activeUserUpdater(store).then();
    setInterval(() => {
        activeUserUpdater(store).then();
    }, 60 * 1000);

    // Do check-in in interval
    const checkIn = () => {
        const state = store.getState();
        if (state.activeUser) {
            usrActivity(state.activeUser?.username!, 10).then();
        }
    }
    checkIn();
    setInterval(checkIn, 1000 * 60 * 15 + 8);

    // Inject / update promoted entries to store
    const promotedEntries = () => {
        getPromotedEntries().then(r => {
            store.dispatch(entriesFetchedAct("__promoted__", r, false));
        })
    }
    promotedEntries();
    setInterval(promotedEntries, 1000 * 60 * 5);

    // Currency
    const currency = ls.get("currency");
    if (currency && currencies.find(x => x.id === currency)) {
        const symbol = getSymbolFromCurrency(currency);
        getCurrencyRate(currency).then(rate => {
            store.dispatch(setCurrency(currency, rate, symbol));
        });
    }

    // Notifications
    if (ls.get("notifications") === false) {
        store.dispatch(muteNotifications());
    }
}
