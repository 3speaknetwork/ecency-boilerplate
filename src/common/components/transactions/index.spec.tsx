import React from "react";

import { TransactionRow } from "./index";
import TestRenderer from "react-test-renderer";
import {
  CurationReward,
  AuthorReward,
  CommentBenefactor,
  ClaimRewardBalance,
  Transfer,
  TransferToVesting,
  WithdrawVesting,
  FillOrder,
} from "../../store/transactions/types";

import { dynamicPropsIntance1 } from "../../helper/test-helper";

jest.mock("moment", () => () => ({
  fromNow: () => "2 hours ago",
}));

it("(1) curation_reward", () => {
  const transaction: CurationReward = {
    comment_author: "user2",
    comment_permlink: "hquwxmvg",
    curator: "user1",
    num: 3204172,
    reward: "33558.958220 VESTS",
    timestamp: "2020-06-06T09:25:12",
    type: "curation_reward",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(2) author_reward", () => {
  const transaction: AuthorReward = {
    author: "user1",
    num: 358711,
    permlink: "lorem-ipsum-dolor",
    hbd_payout: "4.174 HBD",
    hive_payout: "0.000 HIVE",
    timestamp: "2020-06-02T18:02:27",
    type: "author_reward",
    vesting_payout: "31749.102292 VESTS",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(3) comment_benefactor_reward", () => {
  const transaction: CommentBenefactor = {
    author: "xxxthorxxx",
    benefactor: "esteemapp",
    num: 2623508,
    permlink: "re-nonameslefttouse-2020530t135919321z",
    hbd_payout: "0.000 HBD",
    hive_payout: "0.000 HIVE",
    timestamp: "2020-06-06T10:59:24",
    type: "comment_benefactor_reward",
    vesting_payout: "3.893128 VESTS",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(4) comment_benefactor_reward", () => {
  const transaction: ClaimRewardBalance = {
    account: "user1",
    num: 359042,
    reward_hbd: "0.000 HBD",
    reward_hive: "0.000 HIVE",
    reward_vests: "806.030497 VESTS",
    timestamp: "2020-06-04T07:25:33",
    type: "claim_reward_balance",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(5) transfer", () => {
  const transaction: Transfer = {
    amount: "192.425 HIVE",
    from: "user1",
    memo: "lorem-ipsum-dolor",
    num: 3204231,
    timestamp: "2020-06-06T10:07:09",
    to: "user2",
    type: "transfer",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(6) transfer_to_vesting", () => {
  const transaction: TransferToVesting = {
    amount: "82.203 HIVE",
    from: "user1",
    num: 3204242,
    timestamp: "2020-06-06T10:14:33",
    to: "user2",
    type: "transfer_to_vesting",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(7) withdraw_vesting", () => {
  const transaction: WithdrawVesting = {
    num: 3204242,
    timestamp: "2020-06-06T10:14:33",
    acc: "user1",
    vesting_shares: "806.030497 VESTS",
    type: "withdraw_vesting",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("(8) fill_order", () => {
  const transaction: FillOrder = {
    num: 3204242,
    timestamp: "2020-06-06T10:14:33",
    current_pays: "foo",
    open_pays: "bar",
    type: "fill_order",
  };

  const props = {
    dynamicProps: dynamicPropsIntance1,
    transaction,
  };

  const renderer = TestRenderer.create(<TransactionRow {...props} />);
  expect(renderer.toJSON()).toMatchSnapshot();
});
