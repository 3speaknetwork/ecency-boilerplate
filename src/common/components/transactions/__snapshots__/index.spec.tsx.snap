// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) curation_reward 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M15.58,16.8L12,14.5L8.42,16.8L9.5,12.68L6.21,10L10.46,9.74L12,5.8L13.54,9.74L17.79,10L14.5,12.68M20,12C20,10.89 20.9,10 22,10V6C22,4.89 21.1,4 20,4H4A2,2 0 0,0 2,6V10C3.11,10 4,10.9 4,12A2,2 0 0,1 2,14V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V14A2,2 0 0,1 20,12Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Curation Reward
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    17.241 HP
  </div>
  <div
    className="transaction-details"
  >
    <a
      href="/ecency/@user2/hquwxmvg"
      onClick={[Function]}
    >
      @
      user2
      /
      hquwxmvg
    </a>
  </div>
</div>
`;

exports[`(2) author_reward 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M15.58,16.8L12,14.5L8.42,16.8L9.5,12.68L6.21,10L10.46,9.74L12,5.8L13.54,9.74L17.79,10L14.5,12.68M20,12C20,10.89 20.9,10 22,10V6C22,4.89 21.1,4 20,4H4A2,2 0 0,0 2,6V10C3.11,10 4,10.9 4,12A2,2 0 0,1 2,14V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V14A2,2 0 0,1 20,12Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Author Reward
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      4.174 HBD
    </span>
    <span
      className="number"
    >
      16.311 HP
       
    </span>
  </div>
  <div
    className="transaction-details"
  >
    <a
      href="/ecency/@user1/lorem-ipsum-dolor"
      onClick={[Function]}
    >
      @
      user1
      /
      lorem-ipsum-dolor
    </a>
  </div>
</div>
`;

exports[`(3) comment_benefactor_reward 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Comment Benefactor Reward
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      0.002 HP
       
    </span>
  </div>
  <div
    className="transaction-details"
  >
    <a
      href="/ecency/@xxxthorxxx/re-nonameslefttouse-2020530t135919321z"
      onClick={[Function]}
    >
      @
      xxxthorxxx
      /
      re-nonameslefttouse-2020530t135919321z
    </a>
  </div>
</div>
`;

exports[`(4) claim_reward_balance 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M15.58,16.8L12,14.5L8.42,16.8L9.5,12.68L6.21,10L10.46,9.74L12,5.8L13.54,9.74L17.79,10L14.5,12.68M20,12C20,10.89 20.9,10 22,10V6C22,4.89 21.1,4 20,4H4A2,2 0 0,0 2,6V10C3.11,10 4,10.9 4,12A2,2 0 0,1 2,14V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V14A2,2 0 0,1 20,12Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Claim rewards
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      0.414 HP
    </span>
  </div>
  <div
    className="transaction-details"
  />
</div>
`;

exports[`(5) transfer 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M9 14H2V16H9V19L13 15L9 11V14M15 13V10H22V8H15V5L11 9L15 13Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Transfer
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      192.425 HIVE
    </span>
  </div>
  <div
    className="transaction-details"
  >
    <span>
      lorem-ipsum-dolor
       
      <br />
       
      <br />
      <strong>
        @
        user1
      </strong>
       -&gt; 
      <strong>
        @
        user2
      </strong>
    </span>
  </div>
</div>
`;

exports[`(6) transfer_to_vesting 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M9 14H2V16H9V19L13 15L9 11V14M15 13V10H22V8H15V5L11 9L15 13Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Transfer to vesting
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      82.203 HIVE
    </span>
  </div>
  <div
    className="transaction-details"
  >
    <span>
      <strong>
        @
        user1
      </strong>
       -&gt; 
      <strong>
        @
        user2
      </strong>
    </span>
  </div>
</div>
`;

exports[`(7) withdraw_vesting 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,6H21V18H3V6M12,9A3,3 0 0,1 15,12A3,3 0 0,1 12,15A3,3 0 0,1 9,12A3,3 0 0,1 12,9M7,8A2,2 0 0,1 5,10V14A2,2 0 0,1 7,16H17A2,2 0 0,1 19,14V10A2,2 0 0,1 17,8H7Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Withdraw vesting
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      0.414 HP
    </span>
  </div>
  <div
    className="transaction-details"
  >
    <span>
      <strong>
        @
        user1
      </strong>
    </span>
  </div>
</div>
`;

exports[`(8) fill_order 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,15H21V13H3V15M3,19H21V17H3V19M3,11H21V9H3V11M3,5V7H21V5H3Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Fill order
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      foo
       = 
      bar
    </span>
  </div>
  <div
    className="transaction-details"
  />
</div>
`;

exports[`(9) producer_reward 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M14.79,10.62L3.5,21.9L2.1,20.5L13.38,9.21L14.79,10.62M19.27,7.73L19.86,7.14L19.07,6.35L19.71,5.71L18.29,4.29L17.65,4.93L16.86,4.14L16.27,4.73C14.53,3.31 12.57,2.17 10.47,1.37L9.64,3.16C11.39,4.08 13,5.19 14.5,6.5L14,7L17,10L17.5,9.5C18.81,11 19.92,12.61 20.84,14.36L22.63,13.53C21.83,11.43 20.69,9.47 19.27,7.73Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Producer reward
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    0.240 HP
  </div>
  <div
    className="transaction-details"
  />
</div>
`;

exports[`(10) interest 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M5,6H23V18H5V6M14,9A3,3 0 0,1 17,12A3,3 0 0,1 14,15A3,3 0 0,1 11,12A3,3 0 0,1 14,9M9,8A2,2 0 0,1 7,10V14A2,2 0 0,1 9,16H19A2,2 0 0,1 21,14V10A2,2 0 0,1 19,8H9M1,10H3V20H19V22H1V10Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Interest
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      0.570 HBD
    </span>
  </div>
  <div
    className="transaction-details"
  />
</div>
`;

exports[`(11) transfer_to_savings 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M9 14H2V16H9V19L13 15L9 11V14M15 13V10H22V8H15V5L11 9L15 13Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Transfer to savings
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      0.001 HIVE
    </span>
  </div>
  <div
    className="transaction-details"
  >
    <span>
      test memo
       
      <br />
       
      <br />
      <strong>
        @
        talhasch
      </strong>
       -&gt; 
      <strong>
        @
        talhasch
      </strong>
    </span>
  </div>
</div>
`;

exports[`(12) fill_convert_request 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M3,15H21V13H3V15M3,19H21V17H3V19M3,11H21V9H3V11M3,5V7H21V5H3Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Fill convert request
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    <span
      className="number"
    >
      1.507 HBD
       = 
      10.920 HIVE
    </span>
  </div>
  <div
    className="transaction-details"
  />
</div>
`;

exports[`(13) cancel_transfer_from_savings 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Cancel transfer from savings
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  />
  <div
    className="transaction-details"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "From: <strong>@foo</strong> <br /> Request id: <strong>1612448772</strong>",
        }
      }
    />
  </div>
</div>
`;

exports[`(14) return_vesting_delegation 1`] = `
<div
  className="transaction-list-item"
>
  <div
    className="transaction-icon"
  >
    <svg
      viewBox="0 0 24 24"
    >
      <path
        d="M4,11V13H16L10.5,18.5L11.92,19.92L19.84,12L11.92,4.08L10.5,5.5L16,11H4Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    className="transaction-title"
  >
    <div
      className="transaction-name"
    >
      Return vesting delegation
    </div>
    <div
      className="transaction-date"
    >
      2 hours ago
    </div>
  </div>
  <div
    className="transaction-numbers"
  >
    1.965 HP
  </div>
  <div
    className="transaction-details"
  />
</div>
`;

exports[`(30) fallback 1`] = `
<div
  className="transaction-list-item transaction-list-item-raw"
>
  <div
    className="raw-code"
  >
    {"trx_id":"0x14123213","type":"unknown_operation"}
  </div>
</div>
`;
